tx-query/app/src/main.rs

```diff
@@ -26,6 +35,35 @@ fn main() {
             return;
         }
     };
-    // FIXME: it's just a dummy now
-    assert!(check_initchain(enclave.geteid(), 0xab).is_ok());
+
+    // needs to get a lock in enclave_u mod -- FIXME: I guess this is broken if sled is simultaneously opened by tx-validation and tx-query?
```

Comment:
I don't know the usecase here. But, `sled::Db` (at any given `storage_path`) cannot be opened simultaneously by two separate applications.

Reply:
The original planned use case was:

* tx-validation process: opens the sled db and writes sealed TX data into it, as it goes

* (optional) tx-query process: opens the sled db as read-only and uses it to serve decryption query requests

Reply:
but I guess it may need to be re-architected